lib/spack/docs/packaging_guide.rst

```diff
@@ -5476,6 +5724,22 @@ Version Lists
 
 Spack packages should list supported versions with the newest first.
 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Using ``root`` vs ``prefix``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``root`` and ``prefix`` are both attributes that can be queried on a
+package's dependencies, often when passing configure arguments pointing to the
+location of a dependency.  The difference is that while ``prefix`` is the
+location on disk where a concrete package resides, ``root`` is the `logical`
+location that a package resides, which may be different than ``prefix`` in
+the case of virtual packages or other special circumstances.  For most use
+cases inside a package dependency paths should be accessed via
+``self.spec['foo'].root`` instead of ``self.spec['foo'].prefix``.
```

Comment:
Can you explain this, or point to relevant discussion that I might have missed? I think everything we have so far is implemented in terms of `spec.prefix`, so if `spec.root` is preferred this might be a maintenance issue.